Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -796,22 +798,45 @@ extension BrowserViewController: WKNavigationDelegate {
     }
 
     func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        let tab: Browser! = tabManager.getTab(webView)
+        if tab == nil {
```

Comment:
Hm...well, with this commit, I think it's technically possible to hit if we close the tab before the navigation finishes, then the page finishes loading and fires the navigation delegate. That's because this commit attaches the navigation delegate when the tab is added, but never removes the delegate when the tab is removed. I can fix this by setting the tab delegates to `nil` when the tab is removed, and then it should be safe to use ! here.